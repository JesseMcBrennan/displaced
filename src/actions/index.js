export const setSearch = searchResults => ({
  type: 'SET_SEARCH',
  searchResults
})

export const setMessage = createMessage => ({
  type: 'SET_MESSAGE',
  createMessage
})