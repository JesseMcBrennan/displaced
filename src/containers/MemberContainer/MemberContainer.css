body {
  font-family: 'Open Sans', sans-serif;
}

.memberInfo {
  display: flex;
  justify-content: center;
  text-align: center;
  flex-direction: column;
}

.memberInfo input {
  width: 100%;
}

.memberContainer {
  display: flex;
  justify-content: center;
}
