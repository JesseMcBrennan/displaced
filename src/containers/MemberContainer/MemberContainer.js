import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, NavLink } from 'react-router-dom';
import { membersFetch, membersMoreInfo } from '../../utils/fetchCalls.js';
import { membersCleaner } from '../../utils/dataCleaner.js';
import PropTypes from 'prop-types';
import './MemberContainer.css'
import { Card } from '../Card/Card' 

export const MemberContainer = ({ members }) => {
  const displayMembers = members.map((member, index) => {
    if(!members.length) {
      return;
    }
    return (
      <Card
        {...member}
        member={member}
        key={index}
      />
    )
  })
  return(
    <div className="memberContainer">
      <h1>{displayMembers}</h1>
    </div>
  )
}

export const mapStateToProps = state => ({
  members: state.searchResults
})

export default connect(mapStateToProps)(MemberContainer);