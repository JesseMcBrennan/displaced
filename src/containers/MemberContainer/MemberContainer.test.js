import React from 'react';
import ReactDom from 'react-dom';
import { shallow, mount } from 'enzyme';
import { MemberContainer, mapStateToProps } from './MemberContainer'
import { mockMemberData } from '../../utils/mockData/MockMemberData'
import { setSearch } from '../../actions'

describe('MemberContainer container', () => {
  let wrapper; 

  beforeEach(() => {
    wrapper = shallow( 
      <MemberContainer 
        members={mockMemberData}
      />
    )
  })
  it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
  })
  describe('mapStateToProps', () => {
    it('should return an object', () => {
      const mockResult = { members: {} }

      const expected = {}

      const mappedProps = mapStateToProps(mockResult)

      expect(mappedProps).toEqual(expected)

    })
  })
  

})