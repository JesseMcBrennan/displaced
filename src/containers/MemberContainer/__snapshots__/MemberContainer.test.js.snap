// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberContainer container should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    members={
        Array [
            Object {
              "firstName": "Jeff",
              "lastName": "Merkley",
              "memberUrl": "https://www.merkley.senate.gov",
              "nyt": "http://topics.nytimes.com/top/reference/timestopics/people/m/jeff_merkley/index.html",
              "office": "313 Hart Senate Office Building",
              "party": "D",
              "phone": "202-224-3753",
              "title": "Senator, 2nd Class",
              "voting_percentage": 82.92,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        <Card
                history={undefined}
                member={
                        Object {
                                "firstName": "Jeff",
                                "lastName": "Merkley",
                                "memberUrl": "https://www.merkley.senate.gov",
                                "nyt": "http://topics.nytimes.com/top/reference/timestopics/people/m/jeff_merkley/index.html",
                                "office": "313 Hart Senate Office Building",
                                "party": "D",
                                "phone": "202-224-3753",
                                "title": "Senator, 2nd Class",
                                "voting_percentage": 82.92,
                              }
                }
        />
</h1>,
      "className": "memberContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            history={undefined}
            member={
                        Object {
                                    "firstName": "Jeff",
                                    "lastName": "Merkley",
                                    "memberUrl": "https://www.merkley.senate.gov",
                                    "nyt": "http://topics.nytimes.com/top/reference/timestopics/people/m/jeff_merkley/index.html",
                                    "office": "313 Hart Senate Office Building",
                                    "party": "D",
                                    "phone": "202-224-3753",
                                    "title": "Senator, 2nd Class",
                                    "voting_percentage": 82.92,
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "history": undefined,
            "member": Object {
              "firstName": "Jeff",
              "lastName": "Merkley",
              "memberUrl": "https://www.merkley.senate.gov",
              "nyt": "http://topics.nytimes.com/top/reference/timestopics/people/m/jeff_merkley/index.html",
              "office": "313 Hart Senate Office Building",
              "party": "D",
              "phone": "202-224-3753",
              "title": "Senator, 2nd Class",
              "voting_percentage": 82.92,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          <Card
                    history={undefined}
                    member={
                              Object {
                                        "firstName": "Jeff",
                                        "lastName": "Merkley",
                                        "memberUrl": "https://www.merkley.senate.gov",
                                        "nyt": "http://topics.nytimes.com/top/reference/timestopics/people/m/jeff_merkley/index.html",
                                        "office": "313 Hart Senate Office Building",
                                        "party": "D",
                                        "phone": "202-224-3753",
                                        "title": "Senator, 2nd Class",
                                        "voting_percentage": 82.92,
                                      }
                    }
          />
</h1>,
        "className": "memberContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              history={undefined}
              member={
                            Object {
                                          "firstName": "Jeff",
                                          "lastName": "Merkley",
                                          "memberUrl": "https://www.merkley.senate.gov",
                                          "nyt": "http://topics.nytimes.com/top/reference/timestopics/people/m/jeff_merkley/index.html",
                                          "office": "313 Hart Senate Office Building",
                                          "party": "D",
                                          "phone": "202-224-3753",
                                          "title": "Senator, 2nd Class",
                                          "voting_percentage": 82.92,
                                        }
              }
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "history": undefined,
              "member": Object {
                "firstName": "Jeff",
                "lastName": "Merkley",
                "memberUrl": "https://www.merkley.senate.gov",
                "nyt": "http://topics.nytimes.com/top/reference/timestopics/people/m/jeff_merkley/index.html",
                "office": "313 Hart Senate Office Building",
                "party": "D",
                "phone": "202-224-3753",
                "title": "Senator, 2nd Class",
                "voting_percentage": 82.92,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
